gson/src/main/java/com/google/gson/JsonArray.java

```diff
@@ -372,6 +372,15 @@ public boolean getAsBoolean() {
     throw new IllegalStateException();
   }
 
+  /**
+   * Returns this array as List of JsonElement
+   *
+   * @return this array as List of JsonElement
+     */
+  public List<JsonElement> getAsList() {
+    return elements;
```

Comment:
`return Collections.unmodifiableList(new ArrayList<>(elements));`

Reply:
Hm... Why? The idea of this metod is modify JsonArray by modifying List (for example for sorting).

> That said, it seems like an asList does belong here, since this already implements Iterable.
We cannot convert Iterable to List without going over all elements. This is not optimal way.

Reply:
ah, i misunderstood the use case.